### PR TITLE
includes a check whether text is already available.

### PR DESCRIPTION
**Pull request by stillday**

### CODE REVIEW

File: src/lib/components/translater.svelte [hunk 1]
```diff
@@ -13,20 +13,49 @@ let filename = [];
...
+let safeIndex;
+let safeName;
+let safeInput;
 
 // safe input
 const handelClick = async(index, name) => {
-	if (shown[name][index]) safeAtJson(items[index]);	
+	if (shown[name][index]) {
+		if (editorTiny && (tinymce.activeEditor.getContent()).length > 0 ) {
+			message = !message;
```
Comment (dreitzner) [inline src/lib/components/translater.svelte:24]: `showMessage` would be a better name for the variable

File: src/lib/components/translater.svelte [hunk 2]
```diff
@@ -13,20 +13,49 @@ let filename = [];
...
+			safeIndex = index;
+			safeName = name;
+			safeInput = items[index];
+			return
+		} 
+		if (items[index][name]) {
```
Comment (dreitzner) [inline src/lib/components/translater.svelte:30]: Could do `if((items[index][name] || '').length)`

File: src/lib/components/translater.svelte [hunk 3]
```diff
@@ -35,23 +64,57 @@ const handelClick = async(index, name) => {
 		 removed_menuitems: 'newdocument',
 		});
 	}
+	
+}
+
+function dontSafe(entry, index, name) {
+	// if (editorTiny) {
```
Comment (dreitzner) [inline src/lib/components/translater.svelte:71]: Remove unused code?

File: src/lib/components/translater.svelte [hunk 4]
```diff
@@ -35,23 +64,57 @@ const handelClick = async(index, name) => {
...
-async function safeAtJson(entry) {
+function cancelIt() {
+	message = !message;
+}
+
+async function safeAtJson(entry, index, name) {
 	
 	if (editorTiny) {
 		inputHTML = tinymce.activeEditor.getContent();
+		console.log(inputHTML);
```
Comment (dreitzner) [inline src/lib/components/translater.svelte:95]: Remove console log

File: src/app.css [hunk 5]
```diff
@@ -225,3 +246,33 @@ textarea {
 	width: 100%;
 }
 
+/* dialog */
```
Comment (dreitzner) [inline src/app.css:249]: Fix indent for this section

File: src/lib/components/translater.svelte [hunk 6]
```diff
@@ -13,20 +13,49 @@ let filename = [];
...
 let dbjson = [];
+let safeIndex;
+let safeName;
+let safeInput;
 
 // safe input
 const handelClick = async(index, name) => {
-	if (shown[name][index]) safeAtJson(items[index]);	
+	if (shown[name][index]) {
+		if (editorTiny && (tinymce.activeEditor.getContent()).length > 0 ) {
```
Comment (dreitzner) [inline src/lib/components/translater.svelte:23]: You could omit `> 0`

File: src/lib/components/translater.svelte [hunk 7]
```diff
@@ -13,20 +13,49 @@ let filename = [];
...
+			message = !message;
+			safeIndex = index;
+			safeName = name;
+			safeInput = items[index];
```
Comment (dreitzner) [inline src/lib/components/translater.svelte:27]: Maybe you can extract that into a function, as it is used in more than one place

**Review by dreitzner — COMMENTED** [review]
I hope I caught everything

File: src/lib/components/translater.svelte (hunk 3 above)
Comment (stillday) [reply src/lib/components/translater.svelte:79]: has now a comment

File: src/lib/components/translater.svelte (hunk 2 above)
Comment (stillday) [reply src/lib/components/translater.svelte]: is change

File: src/lib/components/translater.svelte (hunk 1 above)
Comment (stillday) [reply src/lib/components/translater.svelte]: sounds better and is cahnge

File: src/lib/components/translater.svelte (hunk 4 above)
Comment (stillday) [reply src/lib/components/translater.svelte]: is remove

File: src/app.css (hunk 5 above)
Comment (stillday) [reply src/app.css:249]: ?

File: src/lib/components/translater.svelte (hunk 6 above)
Comment (stillday) [reply src/lib/components/translater.svelte:23]: its remove.

**Pull request merged**